runners/spark/pom.xml

```diff
@@ -147,6 +147,29 @@
       <scope>provided</scope>
     </dependency>
     <dependency>
+      <groupId>com.esotericsoftware.kryo</groupId>
+      <artifactId>kryo</artifactId>
+      <version>2.21</version>
+      <scope>provided</scope>
+    </dependency>
+    <dependency>
+      <groupId>de.javakaffee</groupId>
+      <artifactId>kryo-serializers</artifactId>
+      <version>0.39</version>
```

Comment:
Shouldn't we depend on version 0.24 since that also has Kryo version 2.21 as a dependency.
Latest 2.X branch of Kryo aligns with version 0.28

I'm not sure if Kryo has made backwards incompatible changes in its APIs which won't appear till runtime because of the "provided" nature.